src/main/webapp/app/exercises/text/assess/text-submission-assessment.component.ts

```diff
@@ -82,6 +86,10 @@ export class TextSubmissionAssessmentComponent extends TextAssessmentBaseCompone
     exerciseDashboardLink: string[];
     isExamMode = false;
 
+    // An observable that can be used to cancel the request for feedback suggestions
+    // if it is not done when the first manual feedback is entered
+    private feedbackSuggestionsObservable: Observable<void>;
```

Comment:
This described cancelation seems to not happen anywhere.

Reply:
Now it does - only on destroy. For the case where manual feedback is added before the response completes, I decided on a different approach (not adding the automatic feedback at all). This is because HTTP fetches cannot be canceled in JS anyway.